R/get_draws.R

```diff
@@ -0,0 +1,444 @@
+      sprintf(
+        " - `$global_rt` with %i rows\n",
+        nrow(x$global_rt)
+      )
+    )
+  }
+  if (length(x$subpop_rt)) {
+    cat(
+      sprintf(
+        " - `$subpop_rt` with %i rows across %i sub-populations\n",
+        nrow(x$subpop_rt),
+        length(unique(x$subpop_rt$subpop))
+      )
+    )
+  }
+
+  cat("You can use $ to access the datasets.\n")
+
+  invisible(x)
+}
+
+#' Constructor for the new_wwinference_fit_draws
+#'
+#' Constructor running some checks on the contents of the data.
+#'
+#' @param predicted_counts Predicted counts
+#' @param predicted_ww Predicted ww concentration
+#' @param global_rt Global Rt()
+#' @param site_level_r Site-level Rt()s
+#' @noRd
+new_wwinference_fit_draws <- function(
+    predicted_counts,
+    predicted_ww,
+    global_rt,
+    subpop_rt) {
+  # Checking colnames: Must match all exactly
+  predicted_counts_colnames <- c(
+    "date", "draw", "observed_value", "pred_value", "total_pop"
+  )
+  if (length(predicted_counts)) {
+    checkmate::assert_names(
+      colnames(predicted_counts),
+      permutation.of = predicted_counts_colnames
+    )
+  }
+
+  predicted_ww_colnames <- c(
+    "below_lod", "date", "draw", "exclude", "flag_as_ww_outlier",
+    "lab", "lab_site_index", "lab_site_name", "log_genome_copies_per_ml",
+    "log_lod", "observed_value", "pred_value", "site", "site_index",
+    "site_pop", "subpop"
+  )
+  if (length(predicted_ww)) {
+    checkmate::assert_names(
+      colnames(predicted_ww),
+      permutation.of = predicted_ww_colnames
+    )
+  }
+
+  global_rt_colnames <- c(
+    "date", "draw", "observed_value", "pred_value", "total_pop"
+  )
+  if (length(global_rt)) {
+    checkmate::assert_names(
+      colnames(global_rt),
+      permutation.of = global_rt_colnames
+    )
+  }
+
+  subpop_rt_colnames <- c(
+    "date", "draw", "pred_value", "site", "site_index", "site_pop",
+    "subpop"
+  )
+  if (length(subpop_rt)) {
+    checkmate::assert_names(
+      colnames(subpop_rt),
+      permutation.of = subpop_rt_colnames
+    )
+  }
+
+  structure(
+    list(
+      predicted_counts = predicted_counts,
+      predicted_ww = predicted_ww,
+      global_rt = global_rt,
+      subpop_rt = subpop_rt
+    ),
+    class = "wwinference_fit_draws"
+  )
+}
+
+#' @export
+#' @rdname get_draws
+#' @param x An object of class `get_draws`.
+#' @param y Ignored in the the case of `plot`.
+#' @details
+#' The plot method for `wwinference_fit_draws` is a wrapper of
+#' `get_plot_forecasted_counts`, `get_plot_ww_conc`, `get_plot_global_rt`,
+#' and `get_plot_subpop_rt`. Depending on the value of `what`, the function
+#' will call the appropriate method.
+#'
+plot.wwinference_fit_draws <- function(x, y = NULL, what, ...) {
+  if (length(what) != 1L) {
+    stop(
+      "The value provided to `what` must be a length one character vector. ",
+      "Currently, it is of length ", length(what), "."
+    )
+  }
+
+  which_what_are_ok <- setdiff(get_draws_what_ok, "all")
```

Comment:
Again, probably want some logic here for the `stan_data_list$include_ww` to tell it to only plot the two outputs we want